lib/demos/reference/motion_demo_shared_x_axis_transition.dart

```diff
@@ -0,0 +1,249 @@
+
+// BEGIN sharedXAxisTransitionDemo
+
+class SharedXAxisTransitionDemo extends StatefulWidget {
+  const SharedXAxisTransitionDemo();
+  @override
+  _SharedXAxisTransitionDemoState createState() =>
+      _SharedXAxisTransitionDemoState();
+}
+
+class _SharedXAxisTransitionDemoState extends State<SharedXAxisTransitionDemo> {
+  bool _isLoggedIn = false;
+
+  void _toggleLoginStatus() {
+    setState(() {
+      _isLoggedIn = !_isLoggedIn;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return Scaffold(
+      resizeToAvoidBottomInset: false,
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(localizations.demoSharedXAxisTitle),
+      ),
+      body: SafeArea(
+        child: Column(
+          children: [
+            Expanded(
+              child: PageTransitionSwitcher(
+                duration: const Duration(milliseconds: 300),
+                reverse: !_isLoggedIn,
+                transitionBuilder: (
+                  child,
+                  animation,
+                  secondaryAnimation,
+                ) {
+                  return SharedAxisTransition(
+                    child: child,
+                    animation: animation,
+                    secondaryAnimation: secondaryAnimation,
+                    transitionType: SharedAxisTransitionType.horizontal,
+                  );
+                },
+                child: _isLoggedIn ? const _CoursePage() : const _SignInPage(),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 15),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  FlatButton(
+                    onPressed: _isLoggedIn ? _toggleLoginStatus : null,
+                    child: Text(localizations.demoSharedXAxisBackButtonText),
+                  ),
+                  RaisedButton(
+                    onPressed: _isLoggedIn ? null : _toggleLoginStatus,
+                    disabledColor: Colors.black12,
+                    child: Text(localizations.demoSharedXAxisNextButtonText),
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _CoursePage extends StatelessWidget {
+  const _CoursePage();
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return ListView(
+      children: [
+        const Padding(
+          padding: EdgeInsets.symmetric(vertical: 8),
+        ),
+        Text(
+          localizations.demoSharedXAxisCoursePageTitle,
+          style: Theme.of(context).textTheme.headline5,
+          textAlign: TextAlign.center,
+        ),
+        const Padding(
+          padding: EdgeInsets.symmetric(vertical: 5),
+        ),
```

Comment:
Prefer SizedBox instead of Padding

```suggestion
        const SizedBox(height:16),
```